Microsoft.Toolkit.Uwp.UI/Extensions/ScrollViewer/ScrollViewerExtensions.MiddleClickScrolling.cs

```diff
@@ -0,0 +1,381 @@
+        }
+
+        private static void CoreWindow_PointerMoved(CoreWindow sender, PointerEventArgs args)
+        {
+            // If condution that occures before scrolling begins
+            if (_isPressed && !_isMoved)
+            {
+                PointerPoint pointerPoint = args.CurrentPoint;
+
+                if (pointerPoint.Properties.IsMiddleButtonPressed)
+                {
+                    _currentPosition = Window.Current.CoreWindow.PointerPosition;
+
+                    var offsetX = _currentPosition.X - _startPosition.X;
+                    var offsetY = _currentPosition.Y - _startPosition.Y;
+
+                    // Settign _isMoved if pointer goes out of threshold value
+                    if (Math.Abs(offsetX) > _threshold || Math.Abs(offsetY) > _threshold)
+                    {
+                        _isMoved = true;
+                    }
+                }
+            }
+
+            // Update current position of the pointer if scrolling started
+            if (CanScroll())
+            {
+                _currentPosition = Window.Current.CoreWindow.PointerPosition;
+            }
+        }
+
+        private static void CoreWindow_PointerReleased(CoreWindow sender, PointerEventArgs args)
+        {
+            // Start deferred moving if the pointer is pressed and not moved
+            if (_isPressed && !_isMoved)
+            {
+                _isDeferredMovingStarted = true;
+
+                // Event to stop deferred scrolling if pointer exited
+                Window.Current.CoreWindow.PointerExited -= CoreWindow_PointerExited;
+                Window.Current.CoreWindow.PointerExited += CoreWindow_PointerExited;
+
+                SetCursorType(0, 0);
+            }
+            else
+            {
+                _isDeferredMovingStarted = false;
+            }
+
+            // Unsubscribe if the pointer is pressed and not DeferredMoving
+            if (_isPressed && !_isDeferredMovingStarted)
+            {
+                UnsubscribeMiddleClickScrolling();
+            }
+        }
+
+        private static void CoreWindow_PointerExited(CoreWindow sender, PointerEventArgs args)
+        {
+            Window.Current.CoreWindow.PointerExited -= CoreWindow_PointerExited;
+            UnsubscribeMiddleClickScrolling();
+        }
+
+        /// <summary>
+        /// Change cursor type depend upon offset from starting position
+        /// </summary>
+        /// <param name="offsetX">Horizontal offset from starting position</param>
+        /// <param name="offsetY">Vertical offset from starting position</param>
+        private static void SetCursorType(double offsetX, double offsetY)
+        {
+            if (!_isCursorAvailable)
+            {
+                return;
+            }
+
+            uint cursorID = 101;
+
+            if (Math.Abs(offsetX) < _threshold && Math.Abs(offsetY) < _threshold)
+            {
+                cursorID = 101;
+            }
+            else
```

Comment:
Can these all be collapsed in the else if, there's only going to be one cursorid in the end, eh?

Not sure if the compiler would optimize, but you could split out the Abs calculation above as you use it in multiple places.